src/django_components/component.py

```diff
@@ -1208,6 +1247,14 @@ def _validate_outputs(self, data: Any) -> None:
         validate_typed_dict(data, data_type, f"Component '{self.name}'", "data")
 
 
+# Perf
```

Comment:
This section below relates to second point in https://github.com/django-components/django-components/pull/910#issuecomment-2599692113:

> Avoid duplicitly creating ComponentNode subclasses, as described in https://github.com/django-components/django-components/pull/910#discussion_r1921047790